### ISSUE TITLE
[iOS] Transition API for transform?

### ISSUE THREAD

**Issue by wcandillon** [software-mansion/react-native-reanimated#321]
I love the new transition API, is there any chance that `transform` property will be supported eventually? Or from the technical side it will be too hard or maybe it doesn't even make sense? 

Alternatively, https://github.com/kmagiera/react-native-reanimated/issues/269 would solve this issue as well.

**Comment by wcandillon** [comment]
This is the kind of animation that could be built with the transition API is it would support `transform`: https://twitter.com/wcandillon/status/1142159471285800962

**Comment by wcandillon** [comment]
I've built a hook that makes it very easy to get animation value from a transition. This hook has been extremely useful for me. I might ship it in redash soon. Btw, I couldn't find the default duration and easing for `Transitionning` has it been documented somewhere?

```js
export const useTransition = (
  state: any,
  src: Animated.Adaptable<number>,
  dest: Animated.Adaptable<number>,
  duration: number = 400,
  easing: Easing = Easing.linear
) => {
  const { transitionVal, clock } = useMemo(
    () => ({
      transitionVal: new Value(0),
      clock: new Clock()
    }),
    []
  );
  useCode(
    set(
      transitionVal,
      runTiming(clock, src, {
        toValue: dest,
        duration,
        easing
      })
    ),
    [state]
  );
  return transitionVal;
};
```

**Comment by wcandillon** [comment]
I noticied that this is an iOS issue only.

**Comment by jakub-gonet** [comment]
We discussed Transitions internally in our team. Considering that v2 makes creating animations easier and Transitions was an experimental API for a long time, we decided to leave Transitions as they are.
 
We won’t actively fix bugs in them, but we’ll happily merge any PR with fixes. We’re also thinking about rewriting Transitions in v2 sometime in the future.

I’m closing this issue, but if you have further questions, I will be happy to help.
